docs(website): add popover, non-modal-dialog-primitive doc pages (#507)
twilio-labs/paste: packages/paste-website/src/pages/components/popover/index.mdx

        depth
        value
      }
    }
  }
`;

<ComponentHeader
  name={props.pageContext.frontmatter.title}
  categoryRoute={SidebarCategoryRoutes.COMPONENTS}
  githubUrl="https://github.com/twilio-labs/paste/tree/master/packages/paste-core/components/popover"
  storybookUrl="https://twilio-labs.github.io/paste/?path=/story/components-popover--default"
  data={props.data.allPasteComponent.edges}
/>

---

<contentwrapper>

<TableOfContents headings={props.data.mdx.headings} />

<content>

<Callout>
  <CalloutTitle>Using alpha components in your product</CalloutTitle>
  <CalloutText>
    Alpha components are ready for you to use in production. Components such as Popover are likely to not see any API
    changes. This is only an alpha due to the lack of documentation and minimal design. Feel free to share your
    experience using this component at #help-design-system or{' '}
    <Anchor href="https://github.com/twilio-labs/paste/discussions/new">start a discussion on GitHub</Anchor>.
  </CalloutText>
</Callout>

## Guidelines

### About Popover

The Popover component is a non-modal-dialog that is commonly used for displaying additional rich content on
top of something.

> 

Reviewer: ```
### Accessibility

A Popover follows the non-modal dialog accessibility guidelines with some enforced opinions.

- There must be a focusable element inside the popover.
- There should be a close button.
- Focus is placed inside the Popover when it it shown
- A Popover is a focus trap
- A Popover must be triggered by an explicit user action, e.g. clicking a button.

Author: I've added that text,

## Examples

### Basic Popover

<LivePreview
  scope={{
    Popover,
    PopoverContainer,
    PopoverButton,
    Text,
  }}
  noInline
>
  {defaultExample}
</LivePreview>

> ### Right Popover

Reviewer: ```suggestion
### Controlling placement
```

import {SidebarCategoryRoutes} from '../../../constants';
import {defaultExample, rightExample} from '../../../component-examples/PopoverExamples';

export const pageQuery = graphql`
  {
    allPasteComponent(filter: {name: {eq: "@twilio-paste/popover"}}) {
      edges {
        node {
          name
          description
          status
          version
        }
      }
    }
    mdx(fields: {slug: {eq: "/popover/"}}) {
      headings {
        depth
        value
      }
    }
  }
`;

<ComponentHeader
  name={props.pageContext.frontmatter.title}
  categoryRoute={SidebarCategoryRoutes.COMPONENTS}
  githubUrl="https://github.com/twilio-labs/paste/tree/master/packages/paste-core/components/popover"
  storybookUrl="https://twilio-labs.github.io/paste/?path=/story/components-popover--default"
  data={props.data.allPasteComponent.edges}
/>

---

<contentwrapper>

<TableOfContents headings={props.data.mdx.headings} />

<content>

<Callout>
  <CalloutTitle>Using alpha components in your product</CalloutTitle>
  <CalloutText>
    Alpha components are ready for you to use in production. Components such as Popover are likely to not see any API

>     changes. This is only an alpha due to the lack of documentation and minimal design. Feel free to share your

Reviewer 2: ```suggestion
    changes. This is only an alpha due to the lack of documentation and a matching Sketch symbol. Feel free to share your
```

    experience using this component by{' '}

>     <Anchor href="https://github.com/twilio-labs/paste/discussions/new">starting a discussion on GitHub</Anchor>.

Reviewer 2: ```suggestion
    <Anchor href="https://github.com/twilio-labs/paste/discussions/new">starting a discussion on GitHub</Anchor>. We are also looking for a contributor to symbolize this component in Sketch. Interested? Reach out!
```

  </CalloutText>
</Callout>

## Guidelines

### About Popover

> The Popover component is a non-modal dialog that is commonly used for displaying additional rich content on

Reviewer 2: ```suggestion
The Popover component is a non-modal dialog that is commonly used for displaying additional rich content
```

> top of something.

Reviewer 2: ```suggestion
above your UI.
```

export const pageQuery = graphql`
  {
    allPasteComponent(filter: {name: {eq: "@twilio-paste/popover"}}) {
      edges {
        node {
          name
          description
          status
          version
        }
      }
    }
    mdx(fields: {slug: {eq: "/popover/"}}) {
      headings {
        depth
        value
      }
    }
  }
`;

<ComponentHeader
  name={props.pageContext.frontmatter.title}
  categoryRoute={SidebarCategoryRoutes.COMPONENTS}
  githubUrl="https://github.com/twilio-labs/paste/tree/master/packages/paste-core/components/popover"
  storybookUrl="https://twilio-labs.github.io/paste/?path=/story/components-popover--default"
  data={props.data.allPasteComponent.edges}
/>

---

<contentwrapper>

<TableOfContents headings={props.data.mdx.headings} />

<content>

<Callout>
  <CalloutTitle>Using alpha components in your product</CalloutTitle>
  <CalloutText>
    Alpha components are ready for you to use in production. Components such as Popover are likely to not see any API
    changes. This is only an alpha due to the lack of documentation and minimal design. Feel free to share your
    experience using this component at #help-design-system or{' '}

>     <Anchor href="https://github.com/twilio-labs/paste/discussions/new">start a discussion on GitHub</Anchor>.

Author: Good call. Making that change here.
